contrib/tracing/log_raw_p2p_msgs.py

```diff
@@ -78,18 +78,23 @@
 int trace_inbound_message(struct pt_regs *ctx) {
     int idx = 0;
     struct p2p_message *msg = msg_arr.lookup(&idx);
+    void *paddr = NULL, *pconn_type = NULL, *pmsg_type = NULL, *pmsg = NULL;
 
     // lookup() does not return a NULL pointer. However, the BPF verifier
     // requires an explicit check that that the `msg` pointer isn't a NULL
     // pointer. See https://github.com/iovisor/bcc/issues/2595
     if (msg == NULL) return 1;
 
     bpf_usdt_readarg(1, ctx, &msg->peer_id);
-    bpf_usdt_readarg_p(2, ctx, &msg->peer_addr, MAX_PEER_ADDR_LENGTH);
-    bpf_usdt_readarg_p(3, ctx, &msg->peer_conn_type, MAX_PEER_CONN_TYPE_LENGTH);
-    bpf_usdt_readarg_p(4, ctx, &msg->msg_type, MAX_MSG_TYPE_LENGTH);
+    bpf_usdt_readarg(2, ctx, &paddr);
+    bpf_probe_read_user_str(&msg->peer_addr, sizeof(msg->peer_addr), paddr);
```

Comment:
Using `bpf_probe_read_user_str` here at least makes sure that only until the zero-terminator is read and not the entire buffer.